src/EnergyPlus/DataEnvironment.cc

```diff
@@ -181,6 +181,122 @@ namespace DataEnvironment {
 
 	// Functions
 
+	// Clears the global data in DataEnvironment.
+	// Needed for unit tests, should not be normally called.
+	void
+	clear_state()
```

Comment:
I think generally most of the namespace variables become member variables of a class and, yes, get recreated on any new object. There may be cases where this needs to be publicly exposed still, but I can't think of any off the top of my head.

Yes it would be nice if each class got instantiated by a data structure parsed by the InputProcessor. I'll address your comment about 'my' parser in a different comment.

I tend to agree with your 3rd comment.